koku/api/user_settings/views.py

```diff
@@ -56,38 +59,43 @@ def __post_init__(self):
         else:
             raise SettingsInvalidFilterException("Invalid not a user setting")
 
-    def update_cost_type(self, request_data, schema_name):
-        serializer = UserSettingUpdateCostTypeSerializer(schema_name, request_data)
+    def update_cost_type(self, schema_name):
+        serializer = UserSettingUpdateCostTypeSerializer(schema_name, self.request.data)
         if serializer.is_valid(raise_exception=True):
-            set_cost_type(schema_name, request_data.get(self.setting))
+            set_cost_type(schema_name, self.request.data.get(self.setting))
             invalidate_view_cache_for_tenant_and_source_type(schema_name, Provider.PROVIDER_AWS)
             return Response(status=status.HTTP_204_NO_CONTENT)
 
-    def update_currency(self, request_data, schema_name):
-        serializer = UserSettingUpdateCurrencySerializer(schema_name, request_data)
+    def update_currency(self, schema_name):
+        serializer = UserSettingUpdateCurrencySerializer(schema_name, self.request.data)
         if serializer.is_valid(raise_exception=True):
-            set_currency(schema_name, request_data.get(self.setting))
+            set_currency(schema_name, self.request.data.get(self.setting))
             invalidate_view_cache_for_tenant_and_all_source_types(schema_name)
             return Response(status=status.HTTP_204_NO_CONTENT)
 
-    def retrieve_user_settings(self, request):
-        users_setting = {"settings": {self.setting: self.get_param(request)}}
+    def retrieve_user_settings(self):
+        users_setting = {"settings": {self.setting: self.get_param(self.request)}}
         users_setting = UserSettingSerializer(users_setting, many=False).data
         return Response(users_setting)
 
 
 class AccountSettings(APIView):
     """Settings views for all user settings."""
 
-    permission_classes = [permissions.AllowAny]
+    def get_permissions(self):
+        if self.request.method.lower() == "get":
+            permission_classes = [DeprecatedSettingsAccessPermission]
```

Comment:
Is there a more precise way to tell when the request is coming from the DDF settings page?  Any indicator in the request? Keying off the request method seems like it could lead to using the incorrect settings class.

Reply:
I couldn't find a more precise way to tell other than the method. However, these are very short lived changes too. The Deprecated class is to allow backwards compatibility, while QE & UI kick the tires on the new settings api & permissions. We will remove the deprecated settings when we release the new settings. I can add it as a comment in our transition plan.

Reply:
Sounds good. Just wanted to pause and consider.